### PR TITLE
Add documentation index file

### PR DESCRIPTION
**Pull request by powerjg**
Signed-off-by: Jason Lowe-Power <jason@lowepower.com>

### CODE REVIEW

File: _pages/documentation/index.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# gem5 Documentation
+
+## Learning gem5
+
+[Learning gem5](learning_gem5/introduction/) gives a prose-heavy introduction to using gem5 for computer architecture research written by Jason Lowe-Power.
+This is a great resource for junior researchers who plan on using gem5 heavily for a research project.
+
+It covers details of how gem5 works starting with [how to create configuration scripts](learning_gem5/simple_config).
+It then goes on to describe [how to modify and extend](learning_gem5/part2) gem5 for your research including [creating `SimObjects`](learning_gem5/part2/helloobject), [using gem5's event-driven simulation infrastructure](learning_gem5/part2/events), and [adding memory system objects](learning_gem5/part2/memoryobject).
+In [Learning gem5 Part 3](learning_gem5/part3) the [Ruby cache coherence model](/documentation/memory_systems/ruby) is discussed in detail including a full implementation of an MSI cache coherence protocol.
```
Comment (BobbyRBruce) [inline _pages/documentation/index.md]: "Learning gem5 Part 3" leads to a very basic index page, which I'm not sure was the original intention: 
Screenshot from 2019-11-21 10-00-53

Ditto the "How to modify and extend" above.

Comment (BobbyRBruce) [reply _pages/documentation/index.md]: Also the "Ruby cache coherence model" link is also broken

File: _pages/documentation/index.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+* CPU models and ISAs
+* Debugging gem5
+* **Your idea here!**
+
+Note: this has been migrated from learning.gem5.org and there are minor problems due to this migration (e.g., missing links, bad formatting).
+Please contact Jason (jason@lowepower.com) or create a PR if you find any errors!
+
+## gem5 101
+
+[gem5 101](gem5_101) is a set of assignments mostly from Wisconsin's graduate computer architecture classes (CS 752, CS 757, and CS 758) which will help you learn to use gem5 for research.
```
Comment (BobbyRBruce) [inline _pages/documentation/index.md]: The gem5_101 link is broken. This is also the case for the `Documentation -> gem5 101` link in the header

File: _pages/documentation/index.md (hunk 1 above)
Comment (powerjg) [reply _pages/documentation/index.md]: That link will be fixed in a follow on patch

Comment (powerjg) [reply _pages/documentation/index.md]: Also, I think I'll keep this basic index page for now. I'll come back (someday) and fill in intros to each section.

File: _pages/documentation/index.md (hunk 2 above)
Comment (powerjg) [reply _pages/documentation/index.md]: Fixed in a different PR. I've rebased this and I think the links works now.

File: _pages/documentation/index.md (hunk 1 above)
Comment (BobbyRBruce) [reply _pages/documentation/index.md]: The "Ruby cache coherence model" link is still broken. Outside of this I'm happy

**Comment by powerjg** [comment]
I believe I addressed the above issues, but TBH, I don't remember ;). Let me know if anything else needs to be updated.

File: _pages/documentation/index.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+# gem5 Documentation
+
+## Learning gem5
+
+[Learning gem5](learning_gem5/introduction/) gives a prose-heavy introduction to using gem5 for computer architecture research written by Jason Lowe-Power.
+This is a great resource for junior researchers who plan on using gem5 heavily for a research project.
+
+It covers details of how gem5 works starting with [how to create configuration scripts](learning_gem5/part1/simple_config).
+It then goes on to describe [how to modify and extend](learning_gem5/part2) gem5 for your research including [creating `SimObjects`](learning_gem5/part2/helloobject), [using gem5's event-driven simulation infrastructure](learning_gem5/part2/events), and [adding memory system objects](learning_gem5/part2/memoryobject).
+In [Learning gem5 Part 3](learning_gem5/part3) the [Ruby cache coherence model](/documentation/memory_systems/ruby) is discussed in detail including a full implementation of an MSI cache coherence protocol.
```
Comment (BobbyRBruce) [inline _pages/documentation/index.md]: This link (Ruby cache coherence model) is still broken.

Comment (powerjg) [reply _pages/documentation/index.md]: Done! Although it points to something from #28. It should work when that's merged.

**Pull request merged**